### PR TITLE
Added solution for Mate HW3.2

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+        if (object == null || object.getClass() != this.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return this.getWidth().equals(rectangle.getWidth())
+                && this.getLength().equals(rectangle.getLength())
+                && this.getColor().equals(rectangle.getColor());
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: there is no need to use getters while you are in a class which has this field

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
...
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * this.getWidth() + 31 * this.getLength() + 31 * this.getColor().hashCode();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: what if any field is null?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Rectangle.java]: Ok, fixed.

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -37,14 +37,14 @@ public boolean equals(Object object) {
...
-                && this.getColor().equals(rectangle.getColor());
+        return this.width.equals(rectangle.width)
+                && this.length.equals(rectangle.length)
+                && this.color.equals(rectangle.color);
     }
 
     @Override
     public int hashCode() {
-        return 31 * this.getWidth() + 31 * this.getLength() + 31 * this.getColor().hashCode();
+        return 31 * this.width + 31 * this.length + 31 * this.color.hashCode();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: what will happen if any of your fields is null?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Rectangle.java]: Also fixed.

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**